irispy/sji.py

```diff
@@ -0,0 +1,188 @@
+import datetime
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+import sunpy.cm as cm
+import sunpy.instr.iris
+import sunpy.map
+import numpy as np
+import matplotlib.colors as colors
+from astropy.io import fits
+'''
+This module provides movie tools for level 2 IRIS SJI fits file
+'''
+
+
+def sji_fits_to_cube(filelist, start=0, stop=None, skip=None, grid=False):
+    """
+    Read SJI files and return a MapCube. Inherits
+    sunpy.instr.iris.SJI_to_cube to stitch multiple sji fits
+    files stitched into one mapcube.
+    Also sets colormap and normalization
+    Assumes hdu index of 0.
+
+
+    Parameters
+    ----------
+    filelist: string or list
+        File to read, if single file string detected, will
+        create a list of length 1.
+
+    start: step
+        Temporal axis index to create MapCube from
+
+    stop: step
+        Temporal index to stop MapCube at
+
+    skip: step
+        Temporal index to skip over MapCube at
+
+
+    Returns
+    -------
+    iris_cube: sunpy.map.MapCube
+        A map cube of the SJI sequence
+    """
+
+    if type(filelist) == str:
+        filelist = [filelist]
+    iris_cube = sunpy.map.MapCube()
+    counter = 1
+    setup = sunpy.instr.iris.SJI_to_cube(filelist[0])
+    xcen = setup[0].meta.get('xcen')
+    ycen = setup[0].meta.get('ycen')
+    time = datetime.datetime.strptime(setup[0].meta.get('startobs'),
+                                      '%Y-%m-%d' + 'T' + '%H:%M:%S.%f')
+    print(xcen, ycen, time)
+    for fname in filelist[0:]:
+        newmap = sunpy.instr.iris.SJI_to_cube(fname)
+        for frame in newmap[start:stop:skip]:
+            if frame.mean() < frame.max():
+                cmap = cm.get_cmap(frame._get_cmap_name())
+                vmax = frame.mean()+3.*frame.std()
+                frame.plot_settings['cmap'] = cmap
+                frame.plot_settings['norm'] = colors.LogNorm(1, vmax)
+                #  todo: iris_intscale
+                resultmapcube.maps.append(frame)
+
+        print(str(counter)+' of '+str(len(filelist)))
+        counter += 1
+    #  todo: pointing correction(rot_hpc)
+
+    #  Option to overlay sun coordinates
+    if grid:
+        iris_cube[0].draw_grid(linestyle='--', color='orange')
+
+    return iris_cube
+
+
+def save_to_mp4(mc, outputfile, fps=60):
```

Comment:
I would suggest refactoring this function so that instead of a mapcube it takes an animation object as input, so you would use it like this:
```python
mc = sunpy.map.Map(files, cube=True)
ani = mc.plot()
save_to_mp4(ani, "myfile.mp4")
```